### ISSUE TITLE
DNS Resolution Failure... Kind of. 

### ISSUE THREAD

**Issue by zombielinux** [tiredofit/docker-db-backup#95]
### Summary

I can set the $DB_HOST variable, and it is passed through to the container correctly (`echo $DB_HOST`). 

If I ping $DB_HOST, it fails until the third attempt. 

Additionally, the container spits out errors constantly, without backing anything up. The error is:

`2022-01-20-02:20:12 [WARN] ** [db-backup] MySQL/MariaDB Server '' is not accessible, retrying.. ( seconds so far)`

Note the missing hostname and apparently "n" seconds so far. 

I'm not sure what the issue is. I'm running this inside docker swarm. 

### Steps to reproduce

1. Set up docker swarm
2. Use example docker-compose.yml
3. Observe error

### What is the expected *correct* behavior?

It should have at least shown the DNS name and incrementing seconds counter. 

### Environment

- Image version / tag: "latest"  
- Host OS: ArchLinux

Log Output:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 00-functions: applying... 
[fix-attrs.d] 00-functions: exited 0.
[fix-attrs.d] 01-s6: applying... 
[fix-attrs.d] 01-s6: exited 0.
[fix-attrs.d] 02-zabbix: applying... 
[fix-attrs.d] 02-zabbix: exited 0.
[fix-attrs.d] 03-logrotate: applying... 
[fix-attrs.d] 03-logrotate: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-startup: executing... 
[cont-init.d] 00-startup: exited 0.
[cont-init.d] 01-timezone: executing... 
[cont-init.d] 01-timezone: exited 0.
[cont-init.d] 02-permissions: executing... 
[cont-init.d] 02-permissions: exited 0.
[cont-init.d] 03-monitoring: executing... 
2022-01-20-02:17:40 [NOTICE] ** [monitoring] Container configured for monitoring with 'zabbix modern'
[cont-init.d] 03-monitoring: exited 0.
[cont-init.d] 04-scheduling: executing... 
2022-01-20-02:17:40 [NOTICE] ** [scheduling] Container configured for scheduled tasks with 'cron'
[cont-init.d] 04-scheduling: exited 0.
[cont-init.d] 05-logging: executing... 
2022-01-20-02:17:40 [NOTICE] ** [logging] Disabling Log Shipping Functionality
[cont-init.d] 05-logging: exited 0.
[cont-init.d] 06-messaging: executing... 
2022-01-20-02:17:40 [NOTICE] ** [messaging] Disabling Container messaging features
[cont-init.d] 06-messaging: exited 0.
[cont-init.d] 10-db-backup: executing... 
[cont-init.d] 10-db-backup: exited 0.
[cont-init.d] 99-container: executing... 
[cont-init.d] 99-container: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
2022-01-20-02:17:41 [STARTING] ** [monitoring] [1] Starting Zabbix Agent (modern) 5.4.9
2022-01-20-02:17:41 [STARTING] ** [scheduling] [1] Starting cron
2022-01-20-02:17:50 [NOTICE] ** [db-backup] Next Backup at 2022-01-20 02:17:50 GMT
2022-01-20-02:18:00 [WARN] ** [db-backup] MySQL/MariaDB Server '' is not accessible, retrying.. ( seconds so far)
2022-01-20-02:18:07 [WARN] ** [db-backup] MySQL/MariaDB Server '' is not accessible, retrying.. ( seconds so far)
2022-01-20-02:18:14 [WARN] ** [db-backup] MySQL/MariaDB Server '' is not accessible, retrying.. ( seconds so far)
2022-01-20-02:18:41 [WARN] ** [db-backup] MySQL/MariaDB Server '' is not accessible, retrying.. ( seconds so far)
2022-01-20-02:18:48 [WARN] ** [db-backup] MySQL/MariaDB Server '' is not accessible, retrying.. ( seconds so far)
```

**Comment by the1ts** [comment]
This is the bug described in #94, i.e. you are using a DB user that works fine for backups, but doesn't follow the pattern added in 2.10.3 to test for correct privileges. 

Fix for now is to revert to previous update 2.10.2 rather than latest until a fix is released.

**Comment by zombielinux** [comment]
> This is the bug described in #94, i.e. you are using a DB user that works fine for backups, but doesn't follow the pattern added in 2.10.3 to test for correct privileges.
> 
> Fix for now is to revert to previous update 2.10.2 rather than latest until a fix is released.

I can see the username being an issue with that bug, but why isn’t the hostname being printed in the debug log?

**Comment by alpar-t** [comment]
2.10.2 solved it for me. I was using the root user to back everything up but was getting the error from the description

**Comment by tiredofit** [comment]
Should be resolved with `2.11.0`

**Comment by the1ts** [comment]
Resolved for me also, the new msyqladmin method for checking liveness works excellent for my perhaps "odd" but working DB privileges.

So don't forget to go back to latest to keep up with all the new functionality, comparability and any fixes.
